nist/curve.go

```diff
@@ -179,10 +179,20 @@ func (p *curvePoint) MarshalBinary() ([]byte, error) {
 }
 
 func (p *curvePoint) UnmarshalBinary(buf []byte) error {
-	p.x, p.y = elliptic.Unmarshal(p.c, buf)
-	if p.x == nil || !p.Valid() {
-		return errors.New("invalid elliptic curve point")
+	// Check whether all bytes after first one are 0, so we
+	// just return the initial point.
```

Comment:
I don't think that the current code below leads to a timing attack (which might happen with byte-by-byte comparisons where secret values are involved) but why take a risk? Maybe do something like the following?
```
var c byte = 0
for _, b := range buf[1:] {
		c |= b
}
if c != 0 {
	p.x, p.y = elliptic.Unmarshal(p.c, buf)
	if p.x == nil || !p.Valid() {
		return errors.New("invalid elliptic curve point")
	}
	return nil
}
...
```
I am probably just nitpicking...

Btw. are there somewhere utility functions in dedis/crypto which do constant time comparisons, like those on https://cryptocoding.net/index.php/Coding_rules#Compare_secret_strings_in_constant_time ? Might come in handy at various points.